### PR TITLE
Check prime number solution and test

### PR DESCRIPTION
**Pull request by ozgurozbekuk**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
  - _It's ok to have extra helper functions if necessary, like with mergesort_
- [ ] The function has correct type annotations
- [ ] The function is not called at the top level of the function file 
  - _Recursive solutions **can** call the function from **inside** the function body_

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in any docstrings or tests
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] There are no commented lines of code
- [ ] There are no `print` statements anywhere
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/check_prime_number.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+def is_prime(number):
+    """
+    Check if a number is prime.
+
+    Parameters:
+    - number (int): The number to check.
+
+    Returns:
+    - bool: True if the number is prime, False otherwise.
+    """
```
Comment (Bikaze) [inline solutions/check_prime_number.py:18]: Include a raised section in the function docstring to account for any assertions

File: solutions/check_prime_number.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    if number <= 1:
+        return False  # Numbers 1 and smaller are not prime
```
Comment (Bikaze) [inline solutions/check_prime_number.py:20]: Try to use an assertion here something like
```
assert number > 1, "Number must be greater than 1"
```

File: solutions/check_prime_number.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    for i in range(2, number):
+        if number % i == 0:  # If a divisor is found, it is not prime
+            return False
+    return True  # If it has no divisors, it is a prime number
+
+
+num = 14  # pylint: disable=invalid-name
+print(f"Is {num} prime? {is_prime(num)}")
```
Comment (Bikaze) [inline solutions/check_prime_number.py:28]: I don't think this is necessary! Is it?

File: solutions/tests/test_check_prime_number.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    Returns:
+    - bool: True if the number is prime, False otherwise.
+    """
+    if number <= 1:
+        return False  # Numbers 1 and smaller are not prime
+    for i in range(2, number):
+        if number % i == 0:  # If a divisor is found, it is not prime
+            return False
+    return True  # If it has no divisors, it is a prime number
```
Comment (Bikaze) [inline solutions/tests/test_check_prime_number.py:51]: Instead of rewriting the function (which is unnecessary duplication try to import it:
```
from ..check_prime_number.py import is_prime
```
But remember please change the file name to be the same as the function name

File: solutions/tests/test_check_prime_number.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+class TestIsPrime(unittest.TestCase):
+    """Test case for the is_prime function."""
+
+    def test_prime(self):
+        """Test prime numbers."""
+        self.assertTrue(is_prime(2))
+        self.assertTrue(is_prime(3))
+        self.assertTrue(is_prime(5))
+        self.assertTrue(is_prime(7))
+        self.assertTrue(is_prime(13))
```
Comment (Bikaze) [inline solutions/tests/test_check_prime_number.py:63]: A test function should have one assertion

File: solutions/tests/test_check_prime_number.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    def test_non_prime(self):
+        """Test non-prime numbers."""
+        self.assertFalse(is_prime(1))
+        self.assertFalse(is_prime(4))
+        self.assertFalse(is_prime(6))
+        self.assertFalse(is_prime(8))
+        self.assertFalse(is_prime(9))
+        self.assertFalse(is_prime(10))
+        self.assertFalse(is_prime(14))
```
Comment (Bikaze) [inline solutions/tests/test_check_prime_number.py:73]: A test function should have one assertion

File: solutions/tests/test_check_prime_number.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+    def test_edge_case(self):
+        """Test edge cases."""
+        self.assertFalse(is_prime(0))
+        self.assertFalse(is_prime(-1))
+        self.assertTrue(is_prime(97))
```
Comment (Bikaze) [inline solutions/tests/test_check_prime_number.py:79]: A test function should have one assertion

File: solutions/check_prime_number.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+"""This function check number for prime or not.
+
+Author: Özgür ÖZBEK
+Date: 11th January 2025
+Group: ET6-foundations-group-16
+"""
+
+
+def is_prime(number):
```
Comment (Bikaze) [inline solutions/check_prime_number.py:9]: The function name should be the same as the file name

**Review by Bikaze — COMMENTED** [review]
Hey ozgur, you doing great. You just need to improve on these few areas.
